src/code/z_fbdemo_circle.c

```diff
@@ -1,19 +1,143 @@
+        this->targetRadius = 0.0f;
+        this->startingRadius = 1.0f;
+    } else {
+        this->startingRadius = 0.0f;
+        this->targetRadius = 1.0f;
+    }
+    this->referenceRadius = this->startingRadius;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_circle/TransitionCircle_SetColor.s")
+void* TransitionCircle_Init(void* thisx) {
+    TransitionCircle* this = (TransitionCircle*)thisx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_circle/TransitionCircle_SetType.s")
+    bzero(this, sizeof(TransitionCircle));
+    this->maskType = 1;
+    this->texture = gCircleTex;
+    this->masks = 6;
+    this->maskt = 6;
+    this->unk_1E = 4;
+    this->unk_1F = 0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_circle/func_80164C14.s")
+    return this;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_circle/TransitionCircle_Draw.s")
+void TransitionCircle_Destroy(void* thisx) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_fbdemo_circle/TransitionCircle_IsDone.s")
+void TransitionCircle_Update(void* thisx) {
+    TransitionCircle* this = (TransitionCircle*)thisx;
+
+    this->isDone = Math_StepToF(&this->referenceRadius, this->targetRadius, this->stepValue);
+}
+
+void TransitionCircle_SetColor(void* thisx, u32 color) {
+    TransitionCircle* this = (TransitionCircle*)thisx;
+
+    this->color.rgba = color;
+}
+
+void TransitionCircle_SetType(void* thisx, s32 type) {
+    TransitionCircle* this = (TransitionCircle*)thisx;
+
+    if (type & TC_SET_PARAMS) {
+        this->maskType = type & 1;
+    } else if (type == 1) {
+        this->direction = TRANSITION_CIRCLE_OUT;
+    } else {
+        this->direction = TRANSITION_CIRCLE_IN;
+    }
+}
+
+void TransitionCircle_LoadAndSetTexture(Gfx** gfxp, TexturePtr texture, s32 fmt, s32 arg3, s32 masks, s32 maskt,
+                                        f32 arg6) {
+    Gfx* gfx = *gfxp;
+    s32 xh = D_801FBBCC;
+    s32 yh = D_801FBBCE;
+    s32 width = 1 << masks;
+    s32 height = 1 << maskt;
+    f32 s;
+    f32 t;
+    s32 dtdy;
+    s32 dsdx;
+
+    gDPLoadTextureBlock_4b(gfx++, texture, fmt, width, height, 0, G_TX_MIRROR | G_TX_CLAMP, G_TX_MIRROR | G_TX_CLAMP,
+                           masks, maskt, G_TX_NOLOD, G_TX_NOLOD);
+    gDPSetTileSize(gfx++, G_TX_RENDERTILE, ((SCREEN_WIDTH / 2) - width) << 2, ((SCREEN_HEIGHT / 2) - height) << 2,
+                   ((SCREEN_WIDTH / 2) + (width - 1)) << 2, ((SCREEN_HEIGHT / 2) + (height - 1)) << 2);
+
+    s = ((1.0f - (1.0f / arg6)) * (SCREEN_WIDTH / 2)) + 70.0f;
+    t = ((1.0f - (1.0f / arg6)) * (SCREEN_HEIGHT / 2)) + 50.0f;
+
+    if (s < -1023.0f) {
+        s = -1023.0f;
+    }
+    if (t < -1023.0f) {
+        t = -1023.0f;
+    }
+
+    if ((s <= -1023.0f) || (t <= -1023.0f)) {
+        dsdx = 0;
+        dtdy = 0;
+    } else {
+        dsdx = ((SCREEN_WIDTH - (2.0f * s)) / gScreenWidth) * 1024.0f;
+        dtdy = ((SCREEN_HEIGHT - (2.0f * t)) / gScreenHeight) * 1024.0f;
```

Comment:
I think these are probably supposed to be << 10, but I suspect floats don't let you. Try it, and if not write it as `* (1 << 10)` instead.